### PR TITLE
[yokurang]: Added a --file flag to accept allow to read the patch from a file as a required command line argument

### PR DESCRIPTION
**Pull request by yokurang**
Hi, I implemented a fix for this issue: https://github.com/panglesd/diffcessible/issues/3

After pulling from this branch, run 

```
dune clean && dune build
```
and execute the binary with the flag to see the results

```
dune exec -- diffcessible --file diff.example
```

### CODE REVIEW

**Review by Julow — COMMENTED** [review]
That was fast ! The code looks very good, thanks !

Though I have a comment.

File: bin/main.ml [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
+let main file_path =
+  let file = In_channel.open_bin file_path in
   let s = In_channel.input_all file in
   let patch = Patch.to_diffs s in
   Interactive_viewer.start patch
 
-open Cmdliner
+let file_arg =
+  let doc = "Path to the file containing the Git diff." in
+  Arg.(required & opt (some string) None & info ["f"; "file"] ~docv:"FILE" ~doc)
```
[the remark this answers is not in the document]
Comment (yokurang) [reply bin/main.ml:12]: Sure thing! I initially implemented it as a positional argument, but then I thought about new users, and if we were to expand the project in the future with potentially new features that also require command line arguments, I thought that having to learn and remember positional arguments might cause some confusion, hence the choice for a flag instead.

What do you think? If you still think a positional argument is still the best choice, I'll make the fix! 

Thanks for the comment!

yokurang

**Review by Julow — APPROVED** [review]
Perfect, thanks ! I'll merge.

File: bin/main.ml (hunk 1 above)
[the remark this answers is not in the document]
Comment (yokurang) [reply bin/main.ml]: FYI @Julow, I just implemented your comment to a positional argument in the meantime. 

to run the changes, do 

```
dune clean && dune build
```

and

```
dune exec -- diffcessible diff.example
```

[the remark this answers is not in the document]
Comment (panglesd) [reply bin/main.ml]: > I thought about new users, and if we were to expand the project in the future with potentially new features that also require command line arguments, I thought that having to learn and remember positional arguments might cause some confusion, hence the choice for a flag instead.

That'a a good point! But usually, optional or "less important" arguments are added as flags, and we don't have to remember the position as long as there is only one positional argument! For instance:

```
diffcessible --colors=true diff.example
```

**Pull request merged**